projects/saturn/src/main/java/io/fairspace/saturn/services/search/AiSearchApp.java

```diff
@@ -0,0 +1,222 @@
+package io.fairspace.saturn.services.search;
+
+import lombok.extern.log4j.*;
+import nl.hyve.llm.LlmConversation;
+
+import org.json.JSONObject;
+
+import spark.Request;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+import io.fairspace.saturn.services.BaseApp;
+
+import static io.fairspace.saturn.auth.RequestContext.getAccessToken;
+import static org.eclipse.jetty.http.MimeTypes.Type.APPLICATION_JSON;
+import static spark.Spark.get;
+import static spark.Spark.post;
+
+import java.io.FileWriter;
+
+@Log4j2
+public class AiSearchApp extends BaseApp {
+    final String CACHE_DIR = "./data/conversations/";
+
+    public AiSearchApp(String basePath) {
+        super(basePath);
+    }
+
+    @Override
+    protected void initApp() {
+        get("/newchat", "application/json", (req, res) -> {
+            try {
+                res.type(APPLICATION_JSON.asString());
+                return new LlmConversation().startChat(getUserKey());
+            } catch (Exception e) {
+                return handleError(req, e);
+            }
+        });
+
+        get("/allconversations", "application/json", (req, res) -> {
+            if (!Files.exists(Paths.get(CACHE_DIR + getUserKey()))) {
+                return new ArrayList<JSONObject>();
+            }
+
+            try (var conversations = Files.list(Paths.get(CACHE_DIR + getUserKey()))) {
+                var content = conversations.map(path -> {
+                    try {
+                        return new String(Files.readAllBytes(path));
+                    } catch (Exception e) {
+                        return "{}";
+                    }
+                }).collect(Collectors.toList());
+
+                var result = new ArrayList<JSONObject>();
+                for (var c : content) {
+                    try {
+                        var obj = new JSONObject(c);
+                        var conversation = new JSONObject();
+                        conversation.put("id", obj.getJSONObject("conversation").getString("conversationId"));
+                        conversation.put("topic", obj.getJSONObject("conversation").getJSONArray("messages").getJSONObject(0).getJSONObject("userInput").getString("input"));
+                        conversation.put("start", new LlmConversation().getStartTime(obj.getJSONObject("conversation").getString("startTime")));
```

Comment:
Please, move all logic to another layer, 'service' not to mixup controller layer with logic and for easier logic testing. 

Also, I would create instance of LlmConversation on the class instance creation instead of creating it on each request